pkg/ingester/ingester.go

```diff
@@ -526,10 +535,16 @@ func (i *Ingester) loop() {
 	}
 }
 
-// ShutdownHandler triggers the following set of operations in order:
+// LegacyShutdownHandler triggers the following set of operations in order:
 //     * Change the state of ring to stop accepting writes.
 //     * Flush all the chunks.
-func (i *Ingester) ShutdownHandler(w http.ResponseWriter, r *http.Request) {
+// Note: This handler does not trigger a termination of the Loki process,
+// despite its name. Instead, the ingester service is stopped, so an external
+// source can trigger a safe termination through a signal to the process.
+// The handler is deprecated and usage is discouraged. Use ShutdownHandler
+// instead.
+func (i *Ingester) LegacyShutdownHandler(w http.ResponseWriter, r *http.Request) {
+	level.Warn(util_log.Logger).Log("msg", "This handler is deprecated and usage is discouraged. Please use /ingester/shutdown?flush=true instead.")
```

Comment:
This message doesn't mention which handler is deprecated

Reply:
True :see_no_evil: